### ISSUE TITLE
[dev] Blob in request body might be corrupted

### ISSUE THREAD

**Issue by frandiox** [cfworker/cfworker#109]
Hi! I'm not sure if this is a bug or I just need to do things differently, so any help or pointers would be appreciated.

I'm trying to upload an image to a third party service (SanityIO) from the worker (frontend => worker => service). This service  accepts `File|Blob|Buffer|ReadableStream` in the body.

I use a normal `<input type="file">` element to get the File blob and pass it in the body of the fetch request. This works when sent directly from the frontend to the service, but I can't seem to get the correct data in the worker when using `cfworker/dev`.

In a node environment, to make it work, you would read the body in a Buffer and pass it to fetch again:

```js
let buffer = []
request.on('data', (chunk) => buffer.push(chunk))

//...

fetch('sanity-api', { body: Buffer.concat(buffer) })
```

However, when changing this code to something similar to what `cfworker/dev` is doing, I'm not sure how to use the request data:

```js
let buffer = ''
request.on('data', (chunk) => { buffer += chunk })

//...

fetch('sanity-api', { body: ??? })
```

I'm not sure how to get an equivalent to the first code block's `body` in the second block. I've tried with `request.blob()`, arrays, etc. but the resulting body is not accepted by the third party service (not a valid file).

Any ideas? Thank you!

**Comment by jdanyow** [comment]
Thanks for reporting this. Fixed in [@cfworker/dev 1.6.10](https://www.npmjs.com/package/@cfworker/dev/v/1.6.10)

**Comment by frandiox** [comment]
Thanks for fixing it. For the record, it's not possible to get the file using `await req.body.arrayBuffer()`.
